internal/pkg/plugin/manager.go

```diff
@@ -68,13 +71,18 @@ func GetManager() *Manager {
 	return _singleton
 }
 
+// Setter method for profile flags
+func (manager *Manager) SetFlag(profFlag string, debugFlag bool) {
+	manager.ProfileFlag = profFlag
```

Comment:
can you change the function input to `profileFlag` for consistency?

Reply:
this seems like a good place to check whether the input profileFlag matches one of the possible values you allow. you can call `log.Fatalf` to output an error message if it doesn't.

Reply:
I check it on line 123. Advising to move it to when the actual flag is assigned instead of where it is below?
`func (manager *Manager) _RegisterShutdownSignalHandlers() {
	var ProfileCleaner interface {
		Stop()
	}
	if (manager.ProfileFlag == "cpu") {
		ProfileCleaner = profile.Start(profile.NoShutdownHook)
	} else if (manager.ProfileFlag == "memory") {
		ProfileCleaner = profile.Start(profile.MemProfile, profile.NoShutdownHook)
	}`

Reply:
added (: